### ISSUE TITLE
Incorrect formula for `estimate_nbands` method

### ISSUE THREAD

**Issue by matthewkuner** [materialsproject/pymatgen#3148]
https://github.com/materialsproject/pymatgen/blob/71864db310569946703bfa8c3f42eaf69c533744/pymatgen/io/vasp/sets.py#L689

According to my version of the VASP source code, in `main.F`, the actual formula for NBANDS should be:

`max((valid_nelect+2)/2 + max(nions/2,3),   valid_nelect*0.6)) + nmag`
with `nmag=0` always set for ISPIN=1 calcs 

Also, `nmag` should be calculated as:
```
nmag = len([m for m in self.incar["MAGMOM"] if not np.allclose(m, 0)])
nmag =(nmag+1)/2    # this step is currently excluded from the code
```

It can later be increased (to be divisible by the number of cores, etc.), but that can't be accounted for without additional context for a given calc.

**Comment by mkhorton** [comment]
This might well be true, but I expect this was written with reference to the [VASP docs](https://www.vasp.at/wiki/index.php/NBANDS). I'm not sure the appropriateness of referring to the source code here.

**Comment by matthewkuner** [comment]
@mkhorton what do you mean? I thought that using the formula from the source code (which is distinct from the docs) would be desirable, as it would closer to the actual NBANDS used by VASP.

**Comment by mkhorton** [comment]
VASP is not an open source code (meaning, a code released under an open source license).

I'm not a lawyer, but generally it's not ok to reference a closed-source code to re-implement a feature into an open-source code, since it violates the terms of the original license.

I don't know how much of that is context dependent; this is something trivial, and is essentially a single line to implement an alternative API to enable use of the original code (anyone using pymatgen for VASP would also need to hold a license for VASP themselves), but nevertheless ...

**Comment by matthewkuner** [comment]
Closed by #3149
